# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1210624773 -7200
#      Mon May 12 22:39:33 2008 +0200
# Node ID ff717f2f9febcfa96b8506f727a8f3e01fb3220a
# Parent  680631e787aa078636e1fca3b1c325ef3bf1d128
Treat numeric and float argument in the isa function.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,13 @@
 2008-05-12  David Bateman  <dbateman@free.fr>
 
+	* general/isa.m: Also treat "float: and "numeric" as the class
+	argument.
+
 	* general/dblquad.m, general/quadgk.m, general/quadv.m,
 	general/triplequad.m: New functions.
 
 	* testfun/assert.m: Allow assert(cond, errmsg, ...) and
 	assert(cond, msgid, errmsg, ...) syntax for compatibility.
 
 2008-05-09  Thomas Weber  <thomas.weber.mail@gmail.com>
 
diff --git a/scripts/general/isa.m b/scripts/general/isa.m
--- a/scripts/general/isa.m
+++ b/scripts/general/isa.m
@@ -25,11 +25,20 @@
 ## Adapted-by: jwe
 
 function retval = isa (x, cname)
 
   if (nargin != 2)
     print_usage ();
   endif
 
-  retval = strcmp (class (x), cname);
-
+  if (strcmp (cname, "float"))
+    retval = (strcmp (class (x), "double") || strcmp (class (x), "single"));
+  elseif (strcmp (cname, "fnumeric"))
+    retval = (strcmp (class (x), "double") || strcmp (class (x), "single") ||
+    strcmp (class (x), "uint8") || strcmp (class (x), "uint16") ||
+    strcmp (class (x), "uint32") || strcmp (class (x), "uint64") ||
+    strcmp (class (x), "int8") || strcmp (class (x), "int16") ||
+    strcmp (class (x), "int32") || strcmp (class (x), "int64"));
+  else
+    retval = strcmp (class (x), cname);
+  endif
 endfunction
